src/Behat/Context/EntityContext.php

```diff
@@ -314,44 +314,82 @@ protected function checkEntityValues($entity, $fields) {
     return $errors;
   }
 
+  /**
+   * Make replacement on given string when this one contains entity-replacement.
+   *
+   * [entity-replacement:user:mail:behat@metadrop.net:uid]
+   *
+   * @param string $value
+   *   String that contains a single or multiple entity-replacement tokens.
+   *
+   * @return string|bool
+   *   Replaced or untouched string or FALSE otherwise.
+   *
+   * @throws \Exception
+   */
+  protected function entityTokensReplacements($value) {
+    $entity_tokens = [];
+    $replacements = [];
+    if (strpos($value, 'entity-replacement') === 0) {
+      $entity_tokens = [$value];
+    }
+    else {
+      $matches = [];
+      preg_match_all('#\[entity-replacement:?([^]]*)\]#', $value, $matches);
+      $entity_tokens = reset($matches);
+    }
+
+    if (empty($entity_tokens)) {
+      return FALSE;
+    }
+
+    // Get entity type list.
+    $entity_types = $this->getCore()->getEntityTypes();
+
+    foreach ($entity_tokens as $entity_token) {
+      $token_pieces = explode(':', str_replace(['[', ']'], ['', ''], $entity_token));
+      array_shift($token_pieces);
+      $entity_type = $token_pieces[0];
+      $field_key = $token_pieces[1];
+      $field_value = $token_pieces[2];
+      $destiny_replacement = $token_pieces[3];
+
+      $keys_exists = isset($entity_type) && isset($field_key) && isset($field_value) && isset($destiny_replacement);
+
+      if (!$keys_exists || !in_array($entity_type, $entity_types)) {
```

Comment:
After https://github.com/Metadrop/behat-contexts/pull/162/files#r1158589986 and https://github.com/Metadrop/behat-contexts/pull/162/files#r1158591469 :
```suggestion
      if (!!in_array($entity_type, $entity_types)) {
```